app/models/manageiq/providers/embedded_ansible/automation_manager/event_catcher/runner.rb

```diff
@@ -1,3 +1,12 @@
 class ManageIQ::Providers::EmbeddedAnsible::AutomationManager::EventCatcher::Runner < ManageIQ::Providers::BaseManager::EventCatcher::Runner
   include ManageIQ::Providers::AnsibleTower::Shared::AutomationManager::EventCatcher::Runner
+
+  def start_event_monitor
+    tid = super
+    return tid unless tid.nil?
+
+    # Get a new copy of the ems record in case the embedded ansible role changed servers
+    @ems.reload
```

Comment:
> - Do we need to reload before the call to super so that whatever being done in super will reflect the new state in ems?

I didn't do it this way because I wanted to stay as close to the original implementation as possible which meant keeping the `@ems` "cache" around and only reloading when it might be necessary. This prevents us from having to go to the database every call (although I'm not sure this method is called very frequently so maybe that doesn't matter).

> - The url is stored in endpoint which will be reloaded as well, right?

It is and it definitely seems that way (I tested this out on live machines).

> - Is this behavior applicable to other managers?

I don't think so because other managers don't expect the endpoint url to change, right?